src/uiUpdater.ts

```diff
@@ -170,21 +173,38 @@ export function loopTasks(selectedProject: Project) {
     }
 
     function renameTaskTitle(task: Task, project: Project){
-        loopProjects()
-        loopTasks(project)
-        let taskTitle = document.querySelector(`.taskTitle${task.id}`)
-        taskTitle?.replaceWith(document.createElement('input'))
-        let inputValue = document.querySelector('input')
+        loopProjects();
+        loopTasks(project);
+        let taskTitle = document.querySelector(`.taskTitle${task.id}`);
+        taskTitle?.replaceWith(document.createElement('input'));
+        let inputValue = document.querySelector('input');
 
         inputValue?.addEventListener('keypress', function(event){
             if(event.key === 'Enter'){
-                if(inputValue?.value)
-                task.title = inputValue?.value
-                loopTasks(project)
+                if(inputValue?.value && inputValue.value.trim() !== ''){
+                    task.title = inputValue?.value;
+                    loopTasks(project);
+                }else{
+                    task.title = task.title;
+                    loopTasks(project);
+                }
             }
         })
 
     }
+        
+    function dateChange(task: Task, project: Project) {
+        let taskDate = document.querySelector(`.taskDueDate${task.id}`)
+        taskDate?.replaceWith(document.createElement('input'))
+        let inputValue = document.querySelector('input')
+        inputValue?.setAttribute('type', 'date')
+        inputValue?.classList.add('h-7')
+        inputValue?.addEventListener('change', function(){
+            if(inputValue?.value)
```

Comment:
does this if do anything? usually they would have `{}` after the statement

Reply:
typescript